src/PrestaShopBundle/EventListener/Admin/Context/ShopContextListener.php

```diff
@@ -183,4 +198,29 @@ private function redirectShopContext(RequestEvent $requestEvent): ?RedirectRespo
             ['setShopContext', 'conf']
         ));
     }
+
+    /**
+     * @throws ReflectionException
+     */
+    private function verifyAllShopAttribute(Request $request): ?ShopConstraint
+    {
+        try {
+            $routeInfo = $this->router->match($request->getPathInfo());
+            $controller = $routeInfo['_controller'];
```

Comment:
```suggestion
            if (empty($routeInfo['_controller'])) {
                return null;
            }

            $controller = $routeInfo['_controller'];
```

Reply:
This should solve the unit tests failing

Reply:
Finally I revert the condition. The test was poorly mocked. Normally you get a `NoConfigurationException` if the route does not match.